src/lib/allocation_tracker.cc

```diff
@@ -140,6 +140,10 @@ DDRes AllocationTracker::init(uint64_t mem_profile_interval,
       static_cast<int>(RingBufferType::kMPSCRingBuffer)) {
     return ddres_error(DD_WHAT_PERFRB);
   }
+  if (track_deallocations) {
+    // 16 times as we want to probability of collision to be low enough
+    _allocated_address_set.init(liveallocation::kMaxTracked * 16);
```

Comment:
Just to make sure we are on same page: the probability of occurring is `k(k+1)/2n` (at startup, but k^2/n is a good generalization when we are running), though probability of individual events is still `k/n`.
What are we optimizing for ? I don't think we want this never to happen.
I would be more interested in keeping memory impact as low as possible for now. The impact of skipping an allocation for live is OK I feel. Skipping it for allocation profiling is not ideal, though we can fix this.

Reply:
Talking more about this, the aim would be to have a stable probability based on maxed tracked.
That means the size would be a function of k^2 (which blows up really fast).
I won't spend too much time on this as I think it might be better to do what @richardstartin was suggesting (make it work for all addresses).